tests/smoke.rs

```diff
@@ -445,6 +458,32 @@ fn dependencies() {
     run!()
 }
 
+/// Test building a dependency specified as `dependencies` in Xargo.toml
+#[cfg(feature = "dev")]
+#[test]
+fn dependencies_from_working_dir() {
+    fn run() -> Result<()> {
+        // need this exact target name to get the right gcc flags
+        const TARGET: &'static str = "thumbv6m-none-eabi";
+
+        let project = Project::new(TARGET)?;
+        project.xargo_toml(
+            r#"
+[dependencies.alloc]
+"#,
+        )?;
+        let td = TempDir::new("xargo_workdir").chain_err(|| "couldn't create a temporary directory")?;
+
+        project.build_from_workdir(TARGET, &td.path())?;
+        assert!(exists("core", TARGET)?);
+        assert!(exists("alloc", TARGET)?);
```

Comment:
What does this have to do with `--manifest-path`?

Reply:
AFAICT there are no targetted tests for cmd line args for me to emulate. I was thinking that if the project built successfully, it meant that the cmdline was being processed correctly. 

If there is an easier way - I would be interested to know.

Reply:
Well, what happens in this test without the rest of your PR?

Reply:
I always delete `~/.xargo` before re-running xargo tests since I think it will store some state there and you want to be sure to get a fresh run.

Reply:
I've added  a `build_outside_project_dir` test which minimally tests the change.